src/pycram/orm/action_designator.py

```diff
@@ -1,18 +1,22 @@
 from typing import Optional
 
-from .base import Base, Position, Quaternion
-import sqlalchemy
+from .base import Base, RobotState, Position, Quaternion
+from .object_designator import ObjectDesignator
 from ..enums import Arms
+from sqlalchemy.orm import Mapped, mapped_column, MappedAsDataclass, relationship
+from sqlalchemy import ForeignKey
 
 
-class Action(Base):
+class Action(MappedAsDataclass, Base):
     """ORM class of pycram.designators.action_designator.ActionDesignator.
     The purpose of this class is to correctly map the inheritance from the action designator class into the database.
     Inheritance is implemented as Joined Table Inheritance (see https://docs.sqlalchemy.org/en/20/orm/inheritance.html)
     """
     __tablename__ = "Action"
-    dtype = sqlalchemy.Column(sqlalchemy.types.String(255))
-    robot_state = sqlalchemy.Column(sqlalchemy.types.Integer, sqlalchemy.ForeignKey("RobotState.id"))
+
+    dtype: Mapped[str] = mapped_column(init=False)
+    robot_state: Mapped[int] = mapped_column(ForeignKey("RobotState.id"), init=False)
+    robot_state_table_entry: Mapped[RobotState] = relationship(init=False)
 
     __mapper_args__ = {
         "polymorphic_identity": __tablename__,
```

Comment:
can polymorphic identities be declared in supertype via sqlalchemy property like the metadata id?